AdrianBarros/assignments/treesDFS/lc113/lc113.py

```diff
@@ -0,0 +1,74 @@
+# Problem Statement #
+
+# Given a binary tree and a number S, find all paths from root-to-leaf such that the sum of all the node values of each path equals S.
+
+# Example
+# -----
+# targetSum = 12
+# Output = [ [ 1, 7, 4 ], [ 1, 9, 2 ] ]
+#      1
+#     / \
+#    7   9
+#  / |   | \
+# 4  5   2  7
+
+
+class TreeNode:
+    def __init__(self, val, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+'''
+input: binary tree
+output: list of paths list  [[]]
+
+step 1: for this problem I am going to use a helper funtion so i can pass the paths array
+step 2: recursively I am going to pass both the left and the right nodes and I will subtract the current value from the sum 
+Step 3: for each path i will also pass an array that will contain the values
+step 4: check if the sum is correct, if it is append the list
+step 5 return the list
+'''
+
```

Comment:
Make sure to note the edge cases in your overview of your approach. Could also be helpful to write more than one test case, especially a case for any edge cases. Also make sure you are taking note of the time and space complexity of your approaches.